lib/rspec/core/hooks.rb

```diff
@@ -503,6 +503,8 @@ def all_hooks_for(position, scope)
         end
 
         def run_owned_hooks_for(position, scope, example_or_group)
+          return if example_or_group.class.respond_to?(:metadata) &&
+                      example_or_group.class.metadata.keys.include?(:skip)
```

Comment:
Actually, one other thing I just realized: the check needs to be `example_or_group.class.metadata[:skip]` because if the user tags a group with `:skip => false` it should _not_ be skipped, so checking for the presence of the `:skip` key is insufficient.